src/node/mempool_args.cpp

```diff
@@ -42,7 +45,12 @@ util::Result<void> ApplyArgsManOptions(const ArgsManager& argsman, const CChainP
 {
     mempool_opts.check_ratio = argsman.GetIntArg("-checkmempool", mempool_opts.check_ratio);
 
-    if (auto mb = argsman.GetIntArg("-maxmempool")) mempool_opts.max_size_bytes = *mb * 1'000'000;
+    if (auto mb = argsman.GetIntArg("-maxmempool")) {
+        if (sizeof(void*) == 4 && *mb > MAX_32BIT_MEMPOOL_MB) {
```

Comment:
Maybe clarify this with something like:

```cpp
constexpr is_32bit{sizeof(void*) == 4};
```

Reply:
we could make it an `if constexpr` as well and use a more descriptive way of checking for 32 bitness:
```suggestion
        if constexpr (SIZE_MAX == UINT32_MAX && *mb > MAX_32BIT_MEMPOOL_MB) {
```

Reply:
Sjors, i took your suggestion. L0rinc, as explained above i don't think this is an improvement.